lib/repl.js

```diff
@@ -852,6 +852,57 @@ ArrayStream.prototype.resume = function() {};
 ArrayStream.prototype.write = function() {};
 
 const requireRE = /\brequire\s*\(['"](([\w@./-]+\/)?(?:[\w@./-]*))/;
+
+
+const fsTabbableMethods = [
+  'access',
+  'accessSync',
+  'chmod',
+  'chmodSync',
+  'chown',
+  'chownSync',
+  'createReadStream',
+  'createWriteStream',
+  'exists',
+  'existsSync',
+  'lchmod',
+  'lchmodSync',
+  'lchown',
+  'lchownSync',
+  'lstat',
+  'lstatSync',
+  'mkdir',
+  'mkdirSync',
+  'open',
+  'openSync',
+  'readdir',
+  'readdirSync',
+  'readFile',
+  'readFileSync',
+  'readlink',
+  'readlinkSync',
+  'realpath',
+  'realpath',
+  'realpathSync',
+  'realpathSync',
+  'rmdir',
+  'rmdirSync',
+  'stat',
+  'statSync',
+  'symlink',
+  'symlinkSync',
+  'truncate',
+  'truncateSync',
+  'unlink',
+  'unlinkSync',
+  'utimes',
+  'utimesSync',
```

Comment:
This *will* become a huge pain to maintain over time, especially since the test also duplicates it.

Reply:
@apapirovski thanks for the review! i thought of exporting it in the `repl` to reduce this maintenance cost, but i was unsure about it. Do you have any other suggestions?

Reply:
@apapirovski @benjamingr An alternative implementation of regex matching is to have a `Symbol` for `tabCompletion` in node. Then each module would add their own implementation of the tab completion on the method level (also allowing non-core modules to enjoy this feature as well).

Reply:
That would actually be awesome and a nice idea.

Reply:
Would checking the exports for functions be enough? The regex then already specifies it must be within a string...

Reply:
I mean I sort of like the idea of a custom `tabCompletion` symbol for `repl` but I don't like the idea of shipping that code to everyone that doesn't even use the `repl` (it's always there on the prototype, for no reason).

Maybe the answer would be to have some sort of a definitions file that `repl` includes that enhances any supported modules. Like `internals/repl/definitions/*.js` (stream, fs, etc.) or something. Just musing out loud...

Reply:
Would it be possible to create the list dynamically? e.g.

```js
Object.keys(fs).filter(name => typeof fs[name] === 'function' && !name.startsWith('_'))
```

That would enable a few methods that aren't currently in this PR (e.g. `writeFileSync`), although it might cause a false positive if there are any methods on `fs` that don't accept a filepath as their first argument.

Personally, I think we could be liberal about enabling tab-completion, and just apply it to all `fs` methods. (By comparison, Bash always tab-completes arguments that look like filepaths, even though it doesn't know the semantics of the command being used.)